downloader/src/main/kotlin/vcs/GitRepo.kt

```diff
@@ -58,6 +58,25 @@ class GitRepo : GitBase() {
                     return super.getInfo().copy(path = manifestFile.relativeTo(workingDir).invariantSeparatorsPath)
                 }
 
+                override fun getNested(): Map<String, VcsInfo> {
+                    val paths = runRepoCommand(workingDir, "list", "-p").stdout.lines().filter { it.isNotBlank() }
+
+                    val nested = mutableMapOf<String, VcsInfo>()
+
+                    paths.forEach { path ->
+                        // Add the nested Repo project.
+                        val workingTree = Git().getWorkingTree(getRootPath().resolve(path))
+                        nested[path] = workingTree.getInfo()
```

Comment:
It just occurred to me we could also write this directly as `GitWorkingTree(getRootPath().resolve(path), gitBase).getInfo()` if you also make `gitBase` protected instead of private in `GitWorkingTree`.

Reply:
No, `gitBase` is initialized with `this` which is `GitRepo` in this case, but I want pure `Git` here.